Kernel/Storage/SD/Registers.h

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Endian.h>
+#include <AK/Types.h>
+
+namespace Kernel::SD {
+
+// SD Host Controller Simplified Specification Version 3.00
+// NOTE: The registers must be 32 bits, because of a quirk in the RPI.
+struct SDRegisters {
+    u32 argument_2;
+    u32 block_size_and_block_count;
+    u32 argument_1;
+    u32 transfer_mode_and_command;
+    u32 response_0;
+    u32 response_1;
+    u32 response_2;
+    u32 response_3;
+    u32 buffer_data_port;
+    u32 present_state;
+    u32 host_configuration_0;
+    u32 host_configuration_1;
+    u32 interrupt_status;
+    u32 interrupt_status_enable;
+    u32 interrupt_signal_enable;
+    u32 host_configuration_2;
+    u32 capabilities_0;
+    u32 capabilities_1;
+    u32 maximum_current_capabilities;
+    u32 maximum_current_capabilities_reserved;
+    u32 force_event_for_auto_cmd_error_status;
+    u32 adma_error_status;
+    u32 adma_system_address[2];
+    u32 preset_value[4];
+    u32 reserved_0[28];
+    u32 shared_bus_control;
+    u32 reserved_1[6];
+    u32 slot_interrupt_status_and_version;
+} __attribute__((packed));
+
+struct OperatingConditionRegister {
+    u32 : 15;
+    u32 vdd_voltage_window_27_28 : 1;
+    u32 vdd_voltage_window_28_29 : 1;
+    u32 vdd_voltage_window_29_30 : 1;
+    u32 vdd_voltage_window_30_31 : 1;
+    u32 vdd_voltage_window_31_32 : 1;
+    u32 vdd_voltage_window_32_33 : 1;
+    u32 vdd_voltage_window_33_34 : 1;
+    u32 vdd_voltage_window_34_35 : 1;
+    u32 vdd_voltage_window_35_36 : 1;
+    u32 switching_to_18v_accepted : 1;
+    u32 : 2;
+    u32 over_2tb_support_status : 1;
+    u32 : 1;
+    u32 uhs2_card_status : 1;
+    u32 card_capacity_status : 1;
+    u32 card_power_up_status : 1;
+
+    static OperatingConditionRegister from_acmd41_response(u32 value)
+    {
+        union {
+            u32 x;
+            struct OperatingConditionRegister ocr;
+        } u;
+        u.x = value;
+        return u.ocr;
+    }
+} __attribute__((packed));
```

Comment:
Why not make this a full union with a `raw` u32 member and an anonymous struct holding all others?